packages/eslint-plugin/src/rules/no-array-delete.ts

```diff
@@ -0,0 +1,84 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import type * as ts from 'typescript';
+
+import { createRule, getParserServices } from '../util';
+
+type MessageId = 'noArrayDelete' | 'useSplice';
+
+export default createRule<[], MessageId>({
+  name: 'no-array-delete',
+  meta: {
+    hasSuggestions: true,
+    type: 'problem',
+    docs: {
+      description: 'Disallow the delete operator with array expressions',
+      recommended: 'strict',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      noArrayDelete:
+        'Using the `delete` operator with an array expression is unsafe.',
+      useSplice: 'Use `{{ target }}.splice({{ key }}, 1)` instead.',
```

Comment:
[Refactor] We don't generally put nodes themselves in messages. Users like predictable messages and having them jump around in length per the nodes breaks that a bit. Also, if the nodes are super long `box.mySuperLongPropertyIUsedToBeAJavaDeveloper("hello", "etc")` then the messages get funky.

Saying "an array splice" or something like that should suffice.

Even better, once the message no longer has node texts in it, the logic to retrieve them can move to inside the `fix` func†ion.

Reply:
that's actually a good reason, done